pkg/asset/logging/filelogging_test.go

```diff
@@ -0,0 +1,145 @@
+			expectedGenerationLog: "",
+		},
+		{
+			name: "test asset with one file",
+			assets: []asset.WritableAsset{
+				&installconfig.InstallConfig{
+					File: &asset.File{
+						Filename: "a.yaml",
+					},
+				},
+			},
+			cmdName:               "test install config",
+			directory:             "test",
+			expectedGenerationLog: "test install config created in: test",
+		},
+		{
+			name: "test asset with two files, same directory",
+			assets: []asset.WritableAsset{
+				&machines.Master{
+					MachineFiles: []*asset.File{
+						getAsset("a.yaml"),
+						getAsset("b.yaml"),
+					},
+				},
+			},
+			cmdName:               "machines",
+			directory:             "test",
+			expectedGenerationLog: "machines created in: test",
+		},
+		{
+			name: "test asset with two files, two directories",
+			assets: []asset.WritableAsset{
+				&machines.Master{
+					MachineFiles: []*asset.File{
+						getAsset("a.yaml"),
+						getAsset("machines/b.yaml"),
+					},
+				},
+			},
+			cmdName:               "machines",
+			directory:             "test",
+			expectedGenerationLog: "machines created in: test and test/machines",
+		},
+		{
+			name: "test asset with two files, two directories, but three entries (same file twice",
+			assets: []asset.WritableAsset{
+				&machines.Master{
+					MachineFiles: []*asset.File{
+						getAsset("a.yaml"),
+						getAsset("machines/b.yaml"),
+						getAsset("machines/b.yaml"),
+					},
+				},
+			},
+			cmdName:               "machines",
+			directory:             "test",
+			expectedGenerationLog: "machines created in: test and test/machines",
+		},
+		{
+			name: "test asset with three files, two directories",
+			assets: []asset.WritableAsset{
+				&machines.Master{
+					MachineFiles: []*asset.File{
+						getAsset("a.yaml"),
+						getAsset("machines/b.yaml"),
+						getAsset("machines/c.yaml"),
+					},
+				},
+			},
+			cmdName:               "machines",
+			directory:             "test/",
+			expectedGenerationLog: "machines created in: test and test/machines",
+		},
+		{
+			name: "test asset with three files, three directories",
+			assets: []asset.WritableAsset{
+				&machines.Master{
+					MachineFiles: []*asset.File{
+						getAsset("a.yaml"),
+						getAsset("machines/b.yaml"),
+						getAsset("control-plane/c.yaml"),
+					},
+				},
+			},
+			cmdName:               "machines",
+			directory:             "test",
+			expectedGenerationLog: "machines created in: test, test/control-plane and test/machines",
+		},
+		{
+			name: "test asset with five files, five directories",
+			assets: []asset.WritableAsset{
+				&machines.Master{
+					MachineFiles: []*asset.File{
+						getAsset("a.yaml"),
+						getAsset("machines/b.yaml"),
+						getAsset("control-plane/c.yaml"),
+						getAsset("master/c.yaml"),
+						getAsset("worker/c.yaml"),
+					},
+				},
+			},
+			cmdName:               "machines",
+			directory:             "test",
+			expectedGenerationLog: "machines created in: test",
+		},
```

Comment:
How could we best handle
```
a.yaml
machines/worker/b.yaml
machines/control-plane/c.yaml
machines/infra.yaml
manifests/d.yaml
```

this current will be `created in: test`
when a better would be `created in:  test, machines, manifests`

Reply:
hmm...sounds interesting and a feature we would want. I would restrict the levels of directory that we check and show only the upper ones. I'll fix that.